x-pack/plugins/cases/public/components/connectors/jira/search_issues.tsx

```diff
@@ -5,105 +5,141 @@
   UseField,
+  useFormData,
 } from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+import { useIsUserTyping } from '../../../common/use_is_user_typing';
 import { useKibana } from '../../../common/lib/kibana';
 import type { ActionConnector } from '../../../../common/types/domain';
 import { useGetIssues } from './use_get_issues';
 import * as i18n from './translations';
 import { useGetIssue } from './use_get_issue';
 
+interface FieldProps {
+  field: FieldHook<string>;
+  options: Array<EuiComboBoxOptionOption<string>>;
+  isLoading: boolean;
+  onSearchComboChange: (value: string) => void;
+}
+
 interface Props {
   actionConnector?: ActionConnector;
-  currentParent: string | null;
 }
 
-const SearchIssuesComponent: React.FC<Props> = ({ actionConnector, currentParent }) => {
-  const [query, setQuery] = useState<string | null>(null);
-  const [selectedOptions, setSelectedOptions] = useState<Array<EuiComboBoxOptionOption<string>>>(
-    []
+const SearchIssuesFieldComponent: React.FC<FieldProps> = ({
+  field,
+  options,
+  isLoading,
+  onSearchComboChange,
+}) => {
+  const { value: parent } = field;
+  const { isInvalid, errorMessage } = getFieldValidityAndErrorMessage(field);
+
+  const selectedOptions = [parent]
+    .map((currentParent: string) => {
+      const selectedParent = options.find((issue) => issue.value === currentParent);
+
+      if (selectedParent) {
+        return selectedParent;
+      }
+
+      return null;
+    })
+    .filter((value): value is EuiComboBoxOptionOption<string> => value != null);
+
+  const onChangeComboBox = (changedOptions: Array<EuiComboBoxOptionOption<string>>) => {
+    field.setValue(changedOptions.length ? changedOptions[0].value ?? '' : '');
+  };
+
+  return (
+    <EuiFormRow
+      id="indexConnectorSelectSearchBox"
+      fullWidth
+      label={i18n.PARENT_ISSUE}
+      isInvalid={isInvalid}
+      error={errorMessage}
+    >
+      <EuiComboBox
+        fullWidth
+        singleSelection
+        async
+        placeholder={i18n.SEARCH_ISSUES_PLACEHOLDER}
+        aria-label={i18n.SEARCH_ISSUES_COMBO_BOX_ARIA_LABEL}
+        isLoading={isLoading}
+        isInvalid={isInvalid}
+        noSuggestions={!options.length}
+        options={options}
+        data-test-subj="search-parent-issues"
+        data-testid="search-parent-issues"
+        selectedOptions={selectedOptions}
+        onChange={onChangeComboBox}
+        onSearchChange={onSearchComboChange}
+      />
+    </EuiFormRow>
   );
+};
+SearchIssuesFieldComponent.displayName = 'SearchIssuesField';
+
+const SearchIssuesComponent: React.FC<Props> = ({ actionConnector }) => {
   const { http } = useKibana().services;
-  const isFirstRender = useRef(true);
+  const [{ fields }] = useFormData<{ fields?: { parent: string } }>({
+    watch: ['fields.parent'],
+  });
+
+  const [query, setQuery] = useState<string | null>(null);
+  const { isUserTyping, onContentChange, onDebounce } = useIsUserTyping();
```

Comment:
I think this will prevent the constant calls to the Jira API whenever the user changes something in the form. (Correct me if I am wrong @cnasikas )

I think this would also fix some issues I got in my ON Week project :D

Reply:
I followed the same pattern as we have with assignees. It is helpful to know when the user is typing. The hook for the Jira API had debouncing logic to avoid constant calls.